chunk/aws_storage_client.go

```diff
@@ -289,6 +336,38 @@ func (d dynamoClientAdapter) UpdateTable(name string, readCapacity, writeCapacit
 	return err
 }
 
+func (a awsStorageClient) GetObject(ctx context.Context, key string) ([]byte, error) {
+	var resp *s3.GetObjectOutput
```

Comment:
If we agree to have storageClient without s3client being set (for table manager), which I would strongly suggest not to do, we need to have at least validation first here in these s3 methods if s3Client is there (: